frontend/src/components/table/schedule/schedule.component.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { ActionModel } from "../../../state/models/action.model";
+import { ApplicationStateModel } from "../../../state/models/application-state.model";
+import { WorkerType } from "../../../state/models/schedule-data/employee-info.model";
+import { ScheduleDataModel } from "../../../state/models/schedule-data/schedule-data.model";
+import { ScheduleDataActionType } from "../../../state/reducers/schedule-data.reducer";
+import { ScheduleRowComponent } from "./schedule-parts/schedule-row.component";
+import { ScheduleActionType } from "./schedule-state.model";
+import "./schedule.component.css";
+import { ChildrenSectionComponent } from "./sections/children-section/children-section.components";
+import { DateSectionComponent } from "./sections/date-section/date-section.component";
+import { ShiftsSectionComponent } from "./sections/shifts-section/shifts-section";
+import { useScheduleState } from "./use-schedule-state";
+
+export function ScheduleComponent() {
+  const scheduleModel = useSelector((state: ApplicationStateModel) => state.scheduleData);
+  const dispatchGlobalState = useDispatch();
+  const [scheduleState, dispatchScheduleState, setNewSchedule, scheduleLogic] = useScheduleState();
+
+  useEffect(() => {
+    if (scheduleState.isScheduleModified && scheduleLogic) {
+      dispatchGlobalState({
```

Comment:
Robimy to przy każdej zmianie (blokowanie, odblokowywanie komórki). Najgorsze jest to, że za każdym razem renderujemy harmonogram od nowa. Musimy to przegadać i znaleść lepsze rozwiązanie